src/net/java/sip/communicator/impl/protocol/jabber/ChatRoomJabberImpl.java

```diff
@@ -3187,6 +3194,72 @@ private void processOtherPresence(Presence presence)
             {
                 member.setStatisticsID(statsId.getStatsId());
             }
+
+            synchronized (chatRoomPresenceListenerExtenders)
+            {
+                chatRoomPresenceListenerExtenders.forEach(extender ->
+                    extender.readPresenceAndModifyMember(presence, member)
+                );
+            }
+        }
+    }
+
+    /**
+     * Enables adding extensions to the presence of the {@link ChatRoom} without
+     * explicitly adding getters and setters for the values stored in these
+     * extensions but instead giving a {@link ChatRoomPresenceListenerExtender}
+     * which handles the logic of reading the values and storing them in the
+     * {@link ChatRoomMemberJabberImpl} by using for example
+     * {@link ChatRoomMemberJabberImpl#setValue(String, Object)}
+     */
+    public interface ChatRoomPresenceListenerExtender
```

Comment:
Please put this into a separate file, i.e. make it a top-level class (interface).